pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/BrokerService.java

```diff
@@ -221,12 +240,27 @@ public BrokerService(PulsarService pulsar) throws Exception {
                 return ObjectMapperFactory.getThreadLocal().readValue(content, HashMap.class);
             }
         };
+        this.blockedDispatchers = new ConcurrentOpenHashSet<>();
         // update dynamic configuration and register-listener
         updateConfigurationAndRegisterListeners();
         this.lookupRequestSemaphore = new AtomicReference<Semaphore>(
                 new Semaphore(pulsar.getConfiguration().getMaxConcurrentLookupRequest(), false));
         this.topicLoadRequestSemaphore = new AtomicReference<Semaphore>(
                 new Semaphore(pulsar.getConfiguration().getMaxConcurrentTopicLoadRequest(), false));
+        if (pulsar.getConfiguration().getUnAckMsgSubscriptionPercentageLimitOnBrokerBlocked() > 0.0) {
+            this.maxUnackedMessages = pulsar.getConfiguration().getMaxUnackedMessagesPerBroker();
+            this.maxUnackedMsgsPerDispatcher = (int) ((maxUnackedMessages
+                    * pulsar.getConfiguration().getUnAckMsgSubscriptionPercentageLimitOnBrokerBlocked()) / 100);
+            log.info("Enabling per-broker unack-message limit {} and dispatcher-limit {} on blocked-broker",
+                    maxUnackedMessages, maxUnackedMsgsPerDispatcher);
+        } else {
+            this.maxUnackedMessages = 0;
+            this.maxUnackedMsgsPerDispatcher = 0;
+            log.warn(
```

Comment:
This could be info. There's nothing "wrong" in setting 0 to disable the max limit